airflow/models.py

```diff
@@ -3024,6 +3024,25 @@ def get_active_runs(self, session=None):
         return active_dates
 
     @provide_session
+    def get_num_active_runs(self, external_trigger=None, session=None):
```

Comment:
I thought about that option, but the following reasons made me not taking that route:
1. `get_active_runs` is actually inconsistent - the name does not reflect what it does, hence it needs to be change; additionally, it's not very useful as it only return execution_dates instead of dag run objects to be more generic and actually use for this purpose as well.
2. Extending `get_active_runs` a little bit would require more thought since the implementation seems very particular to a use case, again, returning only the execution_dates. Thus, this will broad the scope of the PR, which we're trying to avoid.
3. Even though performance seems not to be an issue in this case, having a query that only fetched the numeric count would reduce the number of bits going through the pipe, so that's always good. There are many cases where we don't need the full payload of all active runs, specially when calling the count repeatedly.

Hence, I decided to go agains using `get_active_runs` and implement a more sensitive function: `get_num_active_runs` (renamed after suggestion by @aoen) - Also, seems like this same query could be used in some other parts of the code base, but I think for now to avoid further changes in other places we can avoid them.